boa/src/builtins/number/mod.rs

```diff
@@ -159,6 +159,145 @@ pub fn to_precision(this: &mut Value, args: &[Value], _ctx: &mut Interpreter) ->
+    delta = next_after(0.0, f64::MAX).max(delta);
+    assert!(delta > 0.0);
+    if fraction >= delta {
+        // Insert decimal point.
+        let fresh0 = fraction_cursor;
+        fraction_cursor += 1;
+        buffer[fresh0] = b'.';
+        loop {
+            // Shift up by one digit.
+            fraction *= radix as f64;
+            delta *= radix as f64;
+            // Write digit.
+            let digit = fraction as u32;
+            buffer[fraction_cursor] = std::char::from_digit(digit, radix).unwrap() as u8;
+            fraction_cursor += 1;
+            // Calculate remainder.
+            fraction -= digit as f64;
+            // Round to even.
+            if fraction > 0.5
+                || (fraction - 0.5).abs() < f64::EPSILON && digit & 1 != 0 && fraction + delta > 1.0
+            {
+                loop {
+                    // We need to back trace already written digits in case of carry-over.
+                    fraction_cursor -= 1;
+                    if fraction_cursor == BUF_SIZE / 2 {
+                        //              CHECK_EQ('.', buffer[fraction_cursor]);
+                        // Carry over to the integer part.
+                        integer += 1.;
+                        break;
+                    } else {
+                        let c: u8 = buffer[fraction_cursor];
+                        // Reconstruct digit.
+                        let digit_0 = (c as char).to_digit(10).unwrap();
+                        if digit_0 + 1 >= radix {
+                            continue;
+                        }
+                        buffer[fraction_cursor] =
+                            std::char::from_digit(digit_0 + 1, radix).unwrap() as u8;
+                        fraction_cursor += 1;
+                        break;
+                    }
+                }
+                break;
+            }
+            if fraction < delta {
+                break;
+            }
+        }
+    }
+    // Compute integer digits. Fill unrepresented digits with zero.
+    //  while (Double(integer / radix).Exponent() > 0) {
+    // let mut exp = 0;
+    // unsafe {
+    //     frexp(integer / radix as f64, &mut exp);
+    // }
+
+    while frexp(integer / radix as f64).1 > 0 {
+        integer /= radix as f64;
+        integer_cursor -= 1;
+        buffer[integer_cursor] = b'0';
+    }
+    loop {
+        let remainder = integer % (radix as f64);
+        integer_cursor -= 1;
+        // buffer[integer_cursor as usize] = chars[remainder as libc::c_int as usize];
+        buffer[integer_cursor] = std::char::from_digit(remainder as u32, radix).unwrap() as u8;
+        integer = (integer - remainder) / radix as f64;
+        if integer <= 0f64 {
+            break;
+        }
+    }
+    // Add sign and terminate string.
+    if negative {
+        integer_cursor -= 1;
+        buffer[integer_cursor] = b'-';
+    }
+    assert!(fraction_cursor < BUF_SIZE);
```

Comment:
I think the real bug is here but i am not so sure. The problem is this part is broken from the very beginning.